sympy/solvers/ode.py

```diff
@@ -1334,45 +1335,61 @@ class in it.  Note that a hint may do this anyway if
                 matching_hints["Liouville_Integral"] = r
 
         # Homogeneous second order differential equation of the form
-        # a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3, where
-        # for simplicity, a3, b3 and c3 are assumed to be polynomials.
+        # a3*f(x).diff(x, 2) + b3*f(x).diff(x) + c3
         # It has a definite power series solution at point x0 if, b3/a3 and c3/a3
         # are analytic at x0.
         deq = a3*(f(x).diff(x, 2)) + b3*df + c3*f(x)
         r = collect(reduced_eq,
             [f(x).diff(x, 2), f(x).diff(x), f(x)]).match(deq)
         ordinary = False
+        if r:
+            if not all([r[key].is_polynomial() for key in r]):
+                n, d = reduced_eq.as_numer_denom()
+                reduced_eq = expand(n)
+                r = collect(reduced_eq,
+                    [f(x).diff(x, 2), f(x).diff(x), f(x)]).match(deq)
         if r and r[a3] != 0:
-            if all([r[key].is_polynomial() for key in r]):
```

Comment:
From this condition, If all coeff of` a3*(f(x).diff(x, 2)) + b3*f(x).diff(x) + c3*f(x)` are not polynomials the we are not going for the solution but there is no need of this condition in series solution. 
let's take an example-

```
In [19]: eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 - 4)*f(x)

In [20]: eq
Out[20]: 
     2                                   
 2  d            d          ⎛ 2    ⎞     
x ⋅───(f(x)) + x⋅──(f(x)) + ⎝x  - 4⎠⋅f(x)
     2           dx                      
   dx                                    

In [21]: eq1=expand(eq/x)

In [22]: eq1
Out[22]: 
             2                          
            d          d          4⋅f(x)
x⋅f(x) + x⋅───(f(x)) + ──(f(x)) - ──────
             2         dx           x   
           dx                           
```
When we will pass eq1 then it is not going to solve due to this condition. But a series solution has no problem with any coeff if they are satisfying the defined condition. So  I am removing this condition.